### DISCUSSION TITLE
Subject: ImportError when running a Towhee operator locally

### DISCUSSION THREAD

**Discussion by liaoran321** [towhee-io/towhee#2502] [Q&A]
Hi everyone,

I'm trying to develop a Towhee operator locally, but I'm running into an error that I can't seem to resolve. Whenever I try to run my operator script with python test_add.py, I get an ImportError with the following traceback:

arduino
Copy code
Traceback (most recent call last):
  File "test_add.py", line 2, in <module>
    from towhee.types import DataType, DataShape
ImportError: cannot import name 'DataType' from 'towhee.types' (F:\Towhee\towhee_env\lib\site-packages\towhee\types\__init__.py)
I've checked that Towhee is installed correctly in my virtual environment, and I've followed the steps in the Towhee documentation to create my operator. However, it seems like the DataType and DataShape classes cannot be imported from the towhee.types module.

Here's my operator code in test_add.py:

python
Copy code
from towhee.operator import PyOperator, SharedType
from towhee.types import DataType, DataShape

class TestAdd(PyOperator):
    def __init__(self, factor: int):
        super().__init__()
        self._factor = factor

    def __call__(self, num: int):
        result = self._factor + num
        return result

    @property
    def shared_type(self):
        return SharedType.Shareable

    def input_schema(self):
        return [(DataType.INT32, (1,))]

    def output_schema(self):
        return [(DataType.INT32, (1,))]
Can someone help me figure out what's causing this error? I'd really appreciate any suggestions or advice on how to resolve it.

Thanks in advance!

**Comment by junjiejiangjjj** [comment]
You can refer to this: https://github.com/towhee-io/towhee/tree/main/tests/unittests/mock_operators/add
